### PR TITLE
Dependent Fields New Designs and Place holders

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/fyle/serializers.py [hunk 1]
```diff
@@ -82,6 +83,7 @@ class DependentFieldSettingSerializer(serializers.ModelSerializer):
     project_field_id = serializers.IntegerField(required=False)
     cost_code_field_id = serializers.IntegerField(required=False)
     cost_type_field_id = serializers.IntegerField(required=False)
+    workspace = serializers.CharField(required=False)
```
Comment (ashwin1111) [inline apps/fyle/serializers.py:86]: pls make sure old app doesn't break

File: apps/fyle/serializers.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 from fyle_accounting_mappings.models import ExpenseAttribute
 
+from apps.sage_intacct.dependent_fields import schedule_dependent_field_imports
```
Comment (ashwin1111) [inline apps/fyle/serializers.py:5]: remove unused import

File: apps/fyle/signals.py [hunk 3]
```diff
@@ -36,15 +37,17 @@ def run_pre_save_dependent_field_settings_triggers(sender, instance: DependentFi
         workspace_id=instance.workspace_id,
         fyle_attribute_type=instance.cost_code_field_name,
         platform=platform,
-        parent_field_id=instance.project_field_id
+        source_placeholder=instance.cost_type_placeholder,
```
Comment (ashwin1111) [inline apps/fyle/signals.py:40]: arey bhai, cost_code_place not type

File: apps/fyle/signals.py [hunk 4]
```diff
@@ -56,4 +59,6 @@ def run_post_save_dependent_field_settings_triggers(sender, instance: DependentF
     :param instance: Row instance of Sender Class
     :return: None
     """
-    schedule_dependent_field_imports(instance.workspace_id, instance.is_import_enabled)
+
+    if instance.workspace.app_version == 'v1':
+        async_task('import_dependent_fields_to_fyle', instance.workspace_id)
```
Comment (ashwin1111) [inline apps/fyle/signals.py:64]: path is wrong

File: apps/mappings/signals.py [hunk 5]
```diff
@@ -22,6 +22,7 @@
 from apps.mappings.helpers import schedule_or_delete_fyle_import_tasks
 from apps.tasks.models import Error
 from apps.mappings.models import LocationEntityMapping
+from apps.sage_intacct.dependent_fields import schedule_dependent_field_imports
```
Comment (ashwin1111) [inline apps/mappings/signals.py:25]: rm

File: apps/workspaces/apis/import_settings/serializers.py [hunk 6]
```diff
@@ -85,14 +91,17 @@ class ImportSettingsSerializer(serializers.ModelSerializer):
...
     workspace_id = serializers.SerializerMethodField()
 
+
     class Meta:
         model = Workspace
         fields = [
             'configurations',
             'general_mappings',
             'mapping_settings',
+            'dependent_fields',
```
Comment (ashwin1111) [inline apps/workspaces/apis/import_settings/serializers.py:104]: we should call as per the table name everywhere. It should be `dependent_field_settings`

File: apps/workspaces/apis/import_settings/serializers.py [hunk 7]
```diff
@@ -149,8 +160,15 @@ def update(self, instance, validated):
                     }
                 )
         
+            if configurations_instance.import_projects and dependent_fields:
```
Comment (ashwin1111) [inline apps/workspaces/apis/import_settings/serializers.py:163]: I don't think we're using this flag actively

File: apps/workspaces/apis/import_settings/triggers.py [hunk 8]
```diff
@@ -3,6 +3,8 @@
 
 from apps.mappings.helpers import schedule_or_delete_fyle_import_tasks
 from apps.workspaces.models import Configuration
+from apps.fyle.models import DependentFieldSetting
```
Comment (ashwin1111) [inline apps/workspaces/apis/import_settings/triggers.py:6]: rm

File: apps/workspaces/apis/import_settings/views.py [hunk 9]
```diff
@@ -7,6 +7,8 @@
 
 class ImportSettingsView(generics.RetrieveUpdateAPIView):
     serializer_class = ImportSettingsSerializer
-    
+    authentication_classes = []
+    permission_classes = []
```
Comment (ashwin1111) [inline apps/workspaces/apis/import_settings/views.py:11]: rm

File: apps/workspaces/signals.py [hunk 10]
```diff
@@ -3,11 +3,13 @@
 """
 from django.db.models.signals import post_save
 from django.dispatch import receiver
+from django_q.models import Schedule
 
 from fyle_accounting_mappings.models import MappingSetting
 
 from apps.fyle.helpers import add_expense_id_to_expense_group_settings
 from apps.sage_intacct.helpers import schedule_payment_sync
+from apps.sage_intacct.dependent_fields import schedule_dependent_field_imports
```
Comment (ashwin1111) [inline apps/workspaces/signals.py:12]: rm

File: apps/workspaces/signals.py [hunk 11]
```diff
@@ -3,11 +3,13 @@
 """
 from django.db.models.signals import post_save
 from django.dispatch import receiver
+from django_q.models import Schedule
```
Comment (ashwin1111) [inline apps/workspaces/signals.py:6]: rm

File: tests/test_workspaces/test_apis/test_import_settings/fixtures.py [hunk 12]
```diff
@@ -114,5 +118,6 @@
             "default_tax_code": {"name": "12.5% TR @12.5%", "id": "22"}
         },
         "mapping_settings": None,
+        "dependent_fields": None
```
Comment (ashwin1111) [inline tests/test_workspaces/test_apis/test_import_settings/fixtures.py:121]: there isn't any tests that has dependent fields covered

File: apps/workspaces/apis/import_settings/serializers.py (hunk 7 above)
Comment (NileshPant1999) [reply apps/workspaces/apis/import_settings/serializers.py:177]: what do you mean by actively here @ashwin1111 , i am a bit confused.

**Pull request merged**